src/main.cc

```diff
@@ -178,6 +178,37 @@ signal_cb(ev_loop *loop, struct ev_signal *w, int revents)
 	tarantool_exit(0);
 }
 
+static void
+signal_sigint_cb(ev_loop *loop, struct ev_signal *w, int revents)
+{
+	(void) loop;
+	(void) w;
+	(void) revents;
+
+	/**
+	 * If running in daemon mode, tarantool exits on SIGINT.
+	 * Complain about possibly sudden and unexpected death.
+	 */
+	if (pid_file)
+		say_crit("got signal %d - %s", w->signum, strsignal(w->signum));
+		tarantool_exit(0);
+
+	/**
+	 * Setting prompt explicitly every time, cause of
+	 * need to return from search mode
+	 */
+	rl_set_prompt("tarantool> ");
+
+	ssize_t rc = write(STDOUT_FILENO, "^C\n", sizeof("^C\n")-1);
```

Comment:
I think it is worth adding a constant string with "^C\n" as you a using it two times.

Reply:
Done.